ledger-core/src/core/utils/DerivationPath.cpp

```diff
@@ -47,20 +47,16 @@ namespace ledger {
 
         }
 
-        DerivationPath::DerivationPath(const std::vector<int32_t> &path) : _path(path) {
-            // we cannot use unsigned integer because of some unsupported type in binding 
-            // so we have to check the validity of all the indexes in the path.
-            for (auto index : path)  {
-                assertIndexIsValid(index, "ledger::core::DerivationPath::DerivationPath");
-            }
```

Comment:
I removed those lines because the check is on the index itself and not the value behind that index (I created a JIRA ticket to manage negative derivation value)